app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.ts

```
@@ -0,0 +1,138 @@
+  metaMetricsDataDeletionDate: undefined,
+};
+
+/**
+ * Controller responsible for maintaining
+ * state related to Metametrics data deletion
+ */
+export default class MetaMetricsDataDeletionController {
+  /**
+   * Observable store containing controller data.
+   */
+  store: ObservableStore<MetaMetricsDataDeletionState>;
+
+  private metaMetricsController: MetaMetricsController;
+
+  /**
+   * Constructs a DeleteMetaMetricsData  controller.
+   *
+   * @param options - the controller options
+   * @param options.initState - Initial controller state.
+   * @param options.metaMetricsController
+   */
+  constructor({
+    initState,
+    metaMetricsController,
+  }: {
+    initState: MetaMetricsDataDeletionState;
+    metaMetricsController: MetaMetricsController;
+  }) {
+    this.store = new ObservableStore({
+      ...defaultState,
+      ...initState,
+    });
+    this.metaMetricsController = metaMetricsController;
+  }
+
+  _formatDeletionDate() {
+    const currentDate = new Date();
+    const day = currentDate.getUTCDate();
+    const month = currentDate.getUTCMonth() + 1;
+    const year = currentDate.getUTCFullYear();
+
+    // format the date in the format DD/MM/YYYY
+    return `${day}/${month}/${year}`;
+  }
+
+  async createMetaMetricsDataDeletionTask(): Promise<DataDeletionResponse> {
+    const segmentSourceId = process.env.SEGMENT_DELETE_API_SOURCE_ID;
+    const segmentRegulationEndpoint = process.env.SEGMENT_REGULATIONS_ENDPOINT;
+
+    if (!segmentSourceId || !segmentRegulationEndpoint) {
+      return {
+        status: responseStatus.error,
+        error: 'Segment API source ID or endpoint not found',
+      };
+    }
+    const { metaMetricsId } = this.metaMetricsController.store.getState();
+
+    try {
+      const { data } = await createDataDeletionRegulationTask(
+        metaMetricsId as string,
+      );
+
+      this.store.updateState({
+        metaMetricsDataDeletionId: data?.data?.regulateId as string,
```

If ` data?.data?.regulateId` is `undefined`, should we return an error?

---

I believe the api will return a regulate ID if the request did not error. So I have made the type to just string than `string | undefined`, and initiated the state variable to empty string.